### PR TITLE
[64] tagged animal assessment job feature spec

### LINKED ISSUE #64: Testing - Write Feature/Integration Tests for TaggedAnimalAssessmentJob
Tests need to be written for `TaggedAnimalAssessmentJob`.

Please use this fixture file for testing: `db/sample_data_files/tagged_animal_assessment/Tagged_assessment_12172018 (original).csv`

**Feature/Integration Tests**
The following contexts and expected outcomes should be tested:
1. Context: The user uploads a CSV that has already been processed.
Outcome: 
- A new `ProcessedFile` record should be created 
- On the /file_uploads page, the user should see:
  - File has Status: "Failed"
  - File has Errors: "Already processed a file with the same name. Data not imported!"
  - File has Statistics: "{}"

2. Context: The user uploads a CSV with invalid headers.
Outcome:
- A new `ProcessedFile` record should be created 
- On the /file_uploads page, the user should see:
  - File has Status: "Failed"
  - File has Errors: "Does not have valid headers. Data not imported!"
  - File has Statistics: "{}"

3. Context: The user successfully uploads a CSV with no errors:
Outcome:
- A new `ProcessedFile` record should be created 
- 201 new `TaggedAnimalAssessment` records should be created
- On the /file_uploads page, the user should see:
  - File has Status: "Processed"
  - File has no Errors
  - File has Statistics: "{row_count: 201, rows_imported: 201, rows_not_imported: 0, shl_case_numbers: {"SF16-9A": 100, "SF16-9B": 21, "SF16-9C": 11, "SF16-9D": 69}}"

4. Context: The user successfully uploads a CSV with errors for 2 rows:
Outcome:
- A new `ProcessedFile` record should be created 
- 199 new `TaggedAnimalAssessment` records should be created
- On the /file_uploads page, the user should see:
  - File has Status: "Processed"
  - File has Errors: "Does not have valid headers. Data not imported!"
  - File has Statistics: "{row_count: 201, rows_imported: 199, rows_not_imported: 2, shl_case_numbers: {"SF16-9A": 100, "SF16-9B": 21, "SF16-9C": 11, "SF16-9D": 69}}"

### PR DESCRIPTION
**Pull request by rafaltrojanowski**
Resolves #64  

### Description
- Add Capybara;
- Allow to run DelayedJobs in real-time for tests;
- Add specs as described in the issue (missing one scenario);
- Optimized SpawningSuccessJob#already_processed? method;
- Fixed schema (auto-update after running rake ```db:migrate```);
- Fixed failing test;
- Update ```processed_file``` factory definition;

### Type of change

* Feature spec for existing functionality

### How Has This Been Tested?

By running: rspec spec.

### CODE REVIEW

**Comment by corneliusellen** [comment]
Just saw this, will review this weekend. Thank you for contributing!! :)

**Comment by rafaltrojanowski** [comment]
I would appreciate, thank you.

File: spec/features/upload_tagged_animal_assesments_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          "rows_imported"=>199,
+          "rows_not_imported"=>2,
+        }
+      )
+      expect(page).to have_content expected_success_message
+    end
+  end
+end
+
+# Helper method
```
Comment (corneliusellen) [inline spec/features/upload_tagged_animal_assesments_spec.rb]: Nice.

File: app/models/spawning_success_job.rb [hunk 2]
```diff
@@ -72,7 +72,10 @@ def translate_attribute_names(attrs)
   end
 
   def already_processed?(filename)
-    ProcessedFile.where(status: 'Processed').where(original_filename: original_filename(filename)).count > 0
+    ProcessedFile
+      .where(status: 'Processed')
+      .where(original_filename: original_filename(filename))
+      .exists?
```
Comment (corneliusellen) [inline app/models/spawning_success_job.rb]: Didn't know this - nice optimization. Found an interesting article which explains why: https://semaphoreci.com/blog/2017/03/14/faster-rails-how-to-check-if-a-record-exists.html

Comment (rafaltrojanowski) [reply app/models/spawning_success_job.rb]: ```exists?``` is very handy. There are a few other places where we could apply the same optimization :)

**Pull request merged**